numpyro/svi.py

```diff
@@ -35,6 +35,10 @@ def svi(model, guide, loss, optim_init, optim_update, get_optim_params, **kwargs
     :return: tuple of `(init_fn, update_fn, evaluate)`.
     """
     constrain_fn = None
+    # NB: these will become properties of SVI classes
+    has_dynamic_constraints = False
+    _model_args = None
```

Comment:
`guide_args` and `model_args` might change per mini-batch. Do we need to store this? I think that supporting sample statements with dynamic support is probably more important than supporting dynamic `param` and users will typically have static constraints for these. If so, what do you think about deferring adding this for now? We can mark any failing tests as so and revisit this after the refactor.

Reply:
I think it is particularly unnecessary to support param statements with dynamic constraints. So it is better to remove these stuffs in SVI.